### ISSUE TITLE
Incompatible http dependency in logstash

### ISSUE THREAD

**Issue by moleksyuk** [abonas/kubeclient#206]
Hi @abonas 

I'm developing one logstash plugin with dependency on your `s.add_runtime_dependency 'kubeclient', '~> 2.1.0'` but I can't install it to logstash `2.3.4` because logstash has `http (0.9.9)` and `kubeclient` depends strictly on `spec.add_dependency 'http', '= 0.9.8'`.

```
ERROR: Installation Aborted, message: Bundler could not find compatible versions for gem "http":
  In snapshot (Gemfile.lock):
    http (= 0.9.9)

  In Gemfile:
    logstash-filter-kubernetes_metadata (= 0.1.0) java depends on
      http (= 0.9.8) java
```

Do you plan to update it to newer version or specify not strict dependency?

**Comment by simon3z** [comment]
@moleksyuk would PR #204 solve this issue? Maybe not because @grosser is bumping to `~> 2.0.3`. Can you come up with a common solution that we can evaulate? Thanks.

**Comment by grosser** [comment]
bump http in logstash to be ~> 2.0 instead ... it is unnecessary tighly locked

**Comment by grosser** [comment]
basically they are both doing a silly thing and now they collided :D

**Comment by simon3z** [comment]
@moleksyuk @grosser I am worried about this because bumping a dependency of one (or more :blush: ) major versions is not backward compatible (e.g. what if a current user of this gem has a dependency on `http ~> 0.9`?)

**Comment by grosser** [comment]
then they should bump ... "let us all stay on this old version together" is
not going to work either :D

**Comment by simon3z** [comment]
> then they should bump ... "let us all stay on this old version together" is
> not going to work either :D

Sure but just be aware that this will be scheduled with a major release of kubeclient as well (together with some other non-backward compatible cleanups and changes).

**Comment by grosser** [comment]
sounds like a good plan :)

**Comment by simon3z** [comment]
@grosser @moleksyuk can you update the PR reflecting the decision we made here?

**Comment by cben** [comment]
I believe this is solved.
- `http` was bumped to `~> 2.2.2` on master (#253), released in kubeclient 3.0.0
- `http` was bumped to 1.x or 2.x on kubclient 2.5.2 (#287).  That was actually a mistake, meant to allow 0.9.8 in the stable release, can fix if anyone speaks up.
